tests/default.nix

```diff
@@ -1,20 +1,27 @@
-{ pkgs ? import <nixpkgs> {}, enableBig ? true }:
+{ pkgs ? import <nixpkgs> {}
+, enableBig ? true
+, nmtSrc ? fetchTarball {
+    url =
+      "https://gitlab.com/api/v4/projects/rycee%2Fnmt/repository/archive.tar.gz?sha=d83601002c99b78c89ea80e5e6ba21addcfe12ae";
+    sha256 = "1xzwwxygzs1cmysg97hzd285r7n1g1lwx5y1ar68gwq07a1rczmv";
+  }
+}:
 
 let
 
   lib = import ../modules/lib/stdlib-extended.nix pkgs.lib;
 
-  nmt = fetchTarball {
-    url =
-      "https://gitlab.com/api/v4/projects/rycee%2Fnmt/repository/archive.tar.gz?sha=d83601002c99b78c89ea80e5e6ba21addcfe12ae";
-    sha256 = "1xzwwxygzs1cmysg97hzd285r7n1g1lwx5y1ar68gwq07a1rczmv";
-  };
-
   modules = import ../modules/modules.nix {
     inherit lib pkgs;
     check = false;
   } ++ [
     {
+      # Fix impurities.
+      # The last element of the above `modules` (i.e. `pkgsModule`)
+      # would set `pkgsPath` with `mkDefault` to `<nixpkgs>`
+      # if `config.home.stateVersion` is before 20.09
+      _module.args.pkgsPath = pkgs.path;
+
```

Comment:
Not sure if this should be in an external PR

Reply:
I'll split it out to make it merge faster, considering the discussion to be made about nmd and nmt.